plugins/destination/mongodb/client/migrate.go

```diff
@@ -21,56 +22,53 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
 	return nil
 }
 
-func (c *Client) migrateTable(ctx context.Context, table *schema.Table) error {
+func (c *Client) migrateTable(ctx context.Context, table *arrow.Schema) error {
+	tableName := schema.TableName(table)
 	for _, mdl := range c.getIndexTemplates(table) {
-		res, err := c.client.Database(c.pluginSpec.Database).Collection(table.Name).Indexes().CreateOne(ctx, mdl)
+		res, err := c.client.Database(c.pluginSpec.Database).Collection(tableName).Indexes().CreateOne(ctx, mdl)
 		switch {
 		case err == nil:
-			c.logger.Debug().Str("index_name", res).Str("table", table.Name).Msg("created index")
+			c.logger.Debug().Str("index_name", res).Str("table", tableName).Msg("created index")
 		case isIndexConflictError(err):
-			c.logger.Debug().Str("index_name", res).Str("table", table.Name).Err(err).Msg("create index conflict")
+			c.logger.Debug().Str("index_name", res).Str("table", tableName).Err(err).Msg("create index conflict")
 			if err := c.migrateTableOnConflict(ctx, table, mdl); err != nil {
 				return err
 			}
 		case isIndexOptionsConflictError(err):
-			c.logger.Debug().Str("table", table.Name).Err(err).Msg("skipped create index")
+			c.logger.Debug().Str("table", tableName).Err(err).Msg("skipped create index")
 		default:
-			return fmt.Errorf("create index on %s: %w", table.Name, err)
-		}
-	}
-
-	for _, subTable := range table.Relations {
-		if err := c.migrateTable(ctx, subTable); err != nil {
-			return err
+			return fmt.Errorf("create index on %s: %w", tableName, err)
```

Comment:
I would keep this as-is, wrapping errors causes migrate error messages to get wrapped also
```suggestion
			return err
```